### ISSUE TITLE
Check if Snappy Compressed Stream?

### ISSUE THREAD

**Issue by kevinmeredith** [kesla/node-snappy-stream#1]
How can I check if data is snappy compressed in the Stream format?

https://github.com/kesla/node-snappy provides [isValidSnappyCompressed(input, callback)](https://github.com/kesla/node-snappy#snappyisvalidcompressedinput-callback). 

Perhaps there's something similar for this library?

Thanks

**Comment by kesla** [comment]
No, not currently. Should be fairly simple to do, I'd gladly give you some pointers if you'd like to contribute it. Otherwise I can definitely take a look at it when I have the time :)

**Comment by kevinmeredith** [comment]
Thanks for the quick reply, @kesla.

So, if I had a text file that may or may not be compressed with Snappy Stream, I could **try** to de-compress it. And, if it were not compressed, then an Exception would be thrown? Otherwise, if it were compressed, I could get back the un-compressed text file?

**Comment by kesla** [comment]
Yeah, the stream will emit an error when trying to decompress something that isn't wrong.

TBH, I don't really get why you'd get a stream that might not be streamed? Would be interesting to hear the use case :)
